packages/compass-sidebar/src/components/multiple-connections/active-connections/active-connection-navigation.tsx

```diff
@@ -105,40 +384,81 @@ export function ActiveConnectionNavigation({
     openPerformanceWorkspace,
   } = useOpenWorkspace();
 
+  const [{ filtered, expanded }, dispatch] = useReducer(connectionsReducer, {
+    filtered: undefined,
+    expanded: {},
+  });
+
+  // filter updates
+  // connections change often, but the effect only uses connections if the filter is active
+  // so we use this conditional dependency to avoid too many calls
+  const connectionsButOnlyIfFilterIsActive = filterRegex && connections;
```

Comment:
Should be cast to a boolean if you actually want this to be a primitive value that doesn't trigger updates too often

Reply:
Ah, misread the description, nevermind